### PR TITLE
Ws 6 users delete endpoint

### PR DESCRIPTION
**Pull request by natashabuckham**
Added function Delete(int Id) in UserController.cs
Checked if the user id exists in the database
Deleted the entry in the database
If delete successful, return 200, if not successful return error response
Tested in Swagger

Screenshot 2024-09-09 122130
Screenshot 2024-09-09 122053
Screenshot 2024-09-09 122105

Note: a 404 response happens when user is not found in database
A 500 response happens if a string which isn't a number is passed in the url
Otherwise a 200 response is returned, we decided not to return a response model for this as the frontend has the user id already

### CODE REVIEW

File: backend/Controllers/UserController.cs [hunk 1]
```diff
@@ -21,4 +21,31 @@ public async Task<IActionResult> GetByUserName([FromRoute] string userName)
...
+    public async Task<IActionResult> Delete([FromRoute] string id)
+    {
+        var user = await _userManager.FindByIdAsync(id);
+
+        if (user == null)
+        {
+            return NotFound();
+        }
+
+        var result = await _userManager.DeleteAsync(user);
```
Comment (Kiran-R-K) [inline backend/Controllers/UserController.cs:35]: accessing the database for the delete should be happening in the user service not the controller.
line 28 should also be happening in the user service not the controller - I can see the endpoint above is doing the same thing in the controller (I didn't catch that in a previous review) please could you add methods into the User Service to make each of these database calls and amend the controller to call the?

**Review by Kiran-R-K — CHANGES_REQUESTED** [review]
This looks really good, nice and clean - I left a comment about database calls, let me know if you have any questions

File: backend/Controllers/UserController.cs (hunk 1 above)
Comment (natashabuckham) [reply backend/Controllers/UserController.cs]: All done :)

**Comment by natashabuckham** [comment]
Delete endpoint working with updated code
Screenshot 2024-09-10 120743
Screenshot 2024-09-10 120716

**Pull request merged**